### PR TITLE
deprecate tslint builder and lintFix options

### PR DESCRIPTION
**Pull request by alan-agius4**

### CODE REVIEW

File: packages/angular_devkit/schematics/tasks/tslint-fix/executor.ts [hunk 1]
```diff
@@ -79,7 +79,7 @@ function _listAllFiles(root: string): string[] {
   return result;
 }
 
-
+/** @deprecated Use `ng lint --fix` directly instead. */
```
Comment (alan-agius4) [inline packages/angular_devkit/schematics/tasks/tslint-fix/executor.ts:82]: @kyliau can you check if this is used in Google?

File: packages/angular_devkit/build_angular/src/tslint/index.ts [hunk 2]
```diff
@@ -134,6 +139,7 @@ async function _run(
 }
 
 
+/** @deprecated Deprecated as part of the TSLint deprecation. */
```
Comment (clydin) [inline packages/angular_devkit/build_angular/src/tslint/index.ts:142]: Can you add the version number for the deprecation to the comments as well?

**Review by clydin — APPROVED** [review]

File: packages/angular_devkit/build_angular/src/tslint/index.ts (hunk 2 above)
Comment (alan-agius4) [reply packages/angular_devkit/build_angular/src/tslint/index.ts:142]: Yeah, even in the schema?

File: packages/angular_devkit/build_angular/src/tslint/index.ts [hunk 3]
```diff
@@ -26,6 +26,11 @@ async function _run(
   options: TslintBuilderOptions,
   context: BuilderContext,
 ): Promise<BuilderOutput> {
+  context.logger.warn(
+    'The TSLint builder is deprecated and will be removed in future versions.\n' +
```
Comment (mgechev) [inline packages/angular_devkit/build_angular/src/tslint/index.ts:30]: Let's share why are we deprecating it.

File: packages/angular_devkit/build_angular/src/tslint/index.ts (hunk 2 above)
Comment (clydin) [reply packages/angular_devkit/build_angular/src/tslint/index.ts:142]: probably just the code comments.

File: packages/angular_devkit/schematics/tasks/tslint-fix/executor.ts (hunk 1 above)
Comment (kyliau) [reply packages/angular_devkit/schematics/tasks/tslint-fix/executor.ts:82]: negative. Google doesn't use this task executor to fix code.

Comment (alan-agius4) [reply packages/angular_devkit/schematics/tasks/tslint-fix/executor.ts:82]: Copy that.

File: packages/angular_devkit/build_angular/src/tslint/index.ts (hunk 3 above)
Comment (alan-agius4) [reply packages/angular_devkit/build_angular/src/tslint/index.ts]: @mgechev, PTAL.

**Review by mgechev — APPROVED** [review]

**Pull request merged**